app/lib/api_docs/provider_data.rb

```diff
@@ -61,12 +61,21 @@ def service_ids
       end
     end
 
+    def service_metrics
+      @account.metrics.top_level
+    end
+
+    def backend_api_metrics
+      @account.backend_api_metrics.top_level
+    end
+
     def metrics
-      @metrics ||= @account.metrics.includes(:service).top_level
+      @metrics ||= service_metrics + backend_api_metrics
```

Comment:
You are transforming them into Array though they are scopes.
Maybe you can use Arel::Nodes::Union? 


```suggestion
      @metrics ||= Metric.from(service_metrics.union(backend_api_metrics).to_sql)
```

Reply:
Too complicated to use `Arel::Nodes::Union` here. Exactly like suggested, it results in

```sql
SELECT `metrics`.* FROM ( SELECT `metrics`.* FROM `metrics` INNER JOIN `services` ON `metrics`.`service_id` = `services`.`id` WHERE `services`.`account_id` = ? AND `metrics`.`parent_id` IS NULL ORDER BY `metrics`.`id` ASC UNION SELECT `metrics`.* FROM `metrics` INNER JOIN `backend_apis` ON `metrics`.`owner_id` = `backend_apis`.`id` WHERE `backend_apis`.`account_id` = ? AND `metrics`.`owner_type` = ? AND `metrics`.`parent_id` IS NULL ORDER BY `metrics`.`id` ASC ) ORDER BY `metrics`.`id` ASC"
```

This fails in MySQL in at least 3 different ways:
1. Does not evaluate the binds – need to call `#to_sql` on each component individually or pass the binds explicitly all over again
2. Misuse of `ORDER BY` – could try with `#reorder`, but it starts to complicate stuff too much
3. Missing alias for the inner table (see https://github.com/rails/arel/issues/428)

Reply:
This method is only called by its own class.

![image](https://user-images.githubusercontent.com/11318903/78255446-998b2a80-74f7-11ea-8624-769f5544cb8f.png)

Those 2 classes respond with the 'json' data directly, having done the loop through this just as it is. So, if we make this method `private`, this 'issue' won't matter 😄